docs/developers/piece-reference/i18n.mdx

```diff
@@ -0,0 +1,57 @@
+---
+title: 'Internationalization'
+description: 'Learn about translating pieces to multiple locales'
+icon: 'globe'
+---
+
+
+<Steps>
+  <Step title="Generate The Translation File">
+    After you are done with editing your piece, generate the translation file by running the follwing command:
+    
+    ```bash
+    npm run cli pieces generate-translation-file
```

Comment:
I suggest adding a `generate-i18n` target in your `project.json` in each piece and edit the piece generator CLI to have this target there too.

Then, update the `"build"` target to include `"dependsOn": ["generate-i18n"]`.

Here’s an example: instead of running it as a prebuild step

![image](https://github.com/user-attachments/assets/47e21739-b785-42cd-bc77-a6dfc327f9db)

this target will generate all i18n files by default for every locale.

For each locale file:

* It removes keys that no longer exist.
* It adds any missing keys with default english translation or empty, You should check how crowdin or standard format and how they know if this is already been translated or no.